src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -2280,6 +2280,190 @@ public override IOperation VisitThrow(IThrowOperation operation, int? captureIdF
+                var locals = ArrayBuilder<ILocalSymbol>.GetInstance(declarationGroup.Declarations.Length);
+
+                foreach (IVariableDeclarationOperation declaration in declarationGroup.Declarations)
+                {
+                    foreach (IVariableDeclaratorOperation declarator in declaration.Declarators)
+                    {
+                        locals.Add(declarator.Symbol);
+                        resourceQueue.Add((declaration, declarator));
+                    }
+                }
+
+                resourceQueue.ReverseContents();
+                EnterRegion(new RegionBuilder(ControlFlowGraph.RegionKind.Locals, locals: locals.ToImmutableAndFree()));
+
+                processQueue(resourceQueue);
+
+                LeaveRegion();
+            }
+            else
+            {
+                Debug.Assert(operation.Resources.Kind != OperationKind.VariableDeclaration);
+                Debug.Assert(operation.Resources.Kind != OperationKind.VariableDeclarator);
+
+                // PROTOTYPE(dataflow): Once https://github.com/dotnet/roslyn/issues/25825 is fixed
+                //                      we should handle locals in IUsingOperation.Locals property:
+                //                      'out vars' and the like.
+                IOperation resource = Visit(operation.Resources);
+                int captureId = _availableCaptureId++;
+
+                if (convertToIDisposableBeforeTry(resource))
+                {
+                    resource = convertToIDisposable(resource);
+                }
+
+                AddStatement(new FlowCapture(captureId, resource.Syntax, resource));
+                processResource(new FlowCaptureReference(captureId, resource.Syntax, resource.Type, constantValue: default), resourceQueueOpt: null);
+            }
+
+            return null;
+
+            void processQueue(ArrayBuilder<(IVariableDeclarationOperation, IVariableDeclaratorOperation)> resourceQueueOpt)
+            {
+                if (resourceQueueOpt == null || resourceQueueOpt.Count == 0)
+                {
+                    VisitStatement(operation.Body);
+                }
+                else
+                {
+                    (IVariableDeclarationOperation declaration, IVariableDeclaratorOperation declarator) = resourceQueueOpt.Pop();
+                    HandleVariableDeclarator(declaration, declarator);
+                    ILocalSymbol localSymbol = declarator.Symbol;
+                    processResource(new LocalReferenceExpression(localSymbol, isDeclaration: false, semanticModel: null, declarator.Syntax, localSymbol.Type, 
+                                                                 constantValue: default, isImplicit: true),
+                                    resourceQueueOpt);
+                }
+            }
+
+            bool convertToIDisposableBeforeTry(IOperation resource)
```

Comment:
>convertToIDisposableBeforeTry [](start = 17, length = 29)

Nit: `shouldConvertToIDisposableBeforeTry`. `convert` is a verb, makes me think this will actually do the conversion.